netlab_testing/TLS_Impl_test.cpp

```diff
@@ -0,0 +1,280 @@
+#include "pch.h"
+#include "../netlab/L5/tls_definition.hpp"
+#include "../netlab/L5/tls_protocol_layer.hpp"
+
+using namespace netlab;
+
+class TLS_Tests : public testing::Test {
+public:
+	
+};
+
+TEST_F(TLS_Tests, Serialization_Deserialization_Test_CLIENT_HELLO) {
+
+	HandshakeType msg_type = CLIENT_HELLO;
+
+	TLSHandshakeProtocol handshakeProtocol;
+
+	handshakeProtocol.handshake.updateBody(msg_type);
+
+	handshakeProtocol.handshake.body.clientHello.setClientHello();
+
+	std::string serialized_string = handshakeProtocol.serialize_handshake_protocol_data(msg_type);
+
+	handshakeProtocol.deserialize_handshake_protocol_data(serialized_string, msg_type);
+
+	std::string serialized_string_2 = handshakeProtocol.serialize_handshake_protocol_data(msg_type);
+
+	ASSERT_EQ(serialized_string, serialized_string_2);
```

Comment:
i think this is not perfect.
if you are consistent in the error, like byte order, in both serialize and de-esrlisdfsad you will pass the test but fail in a real communication

Reply:
I might not understand the question.
I check serialization/deserialization. By the design of the test, if the object is assembled into a string and disassembled back apart into the same object - we passed.  I checked every byte in the result, what do you think may not be covered?